### ISSUE TITLE
Remove a cloned subrepo

### ISSUE THREAD

**Issue by callalilychen** [ingydotnet/git-subrepo#239]
Hi,

what is the better way to remove a subrepo reference, after cloning a repository using subrepo?

I tried to do it with git-subrepo clean, and I also removed the directory and the item in .git/config. But I can still see the whole history of the clone sub-repository, using gitk, even if `git-subrepo status` told me, that I have no subrepos.

**Comment by grimmySwe** [comment]
If you remove the subrepo/.gitrepo file you remove everything. Note that when you commit things in your regular repo (subrepo pull) they will stay.

Could you give some more examples so I understand what you mean?

**Comment by callalilychen** [comment]
for example:
I have "testproject" as my regular repo and a "subrepotest" repo. I cloned "subrepotest" into "testproject" using git-subrepo clone.

Then my .git/config is as following:

```
cat .git/config 
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
[remote "subrepo/subrepotest"]
	url = ../subrepotest
	fetch = +refs/heads/*:refs/remotes/subrepo/subrepotest/
```

Now I would like to remove the "subrepotest" from my regular repo, so I do `git-subrepo clean --ALL`.

Then I get a change that subrepotest./.gitrepo ist now deleted. But I can still see the whole history of "subrepotest" with gitk. My `git log` shows also the information of the removed subrepo information:

```
33mcommit b1c3476079e4b8d1af4861583693d57b8adbb8a2m
Author: wenwen <wenwen-git@xxx.de>
Date:	Wed Nov 23 21:34:28 2016 +0100

    delete subrepotest

33mcommit 66ad1a993bd5888080ca1fb59aa8724c2a5cdd0am
Author: wenwen <wenwen-git@xxx.de>
Date:	Wed Nov 23 21:26:34 2016 +0100

    git subrepo clone ../subrepotest

    subrepo:
      subdir:	"subrepotest"
      merged:	"65937f9"
    upstream:
      origin:	"../subrepotest"
      branch:	"master"
      commit:	"65937f9"
    git-subrepo:
      version:	"0.3.0"
      origin:	"git@github.com:ingydotnet/git-subrepo.git"
      commit:	"c162eb1"

33mcommit 168f7493c91ee9f635c4ece1e969605fc4e2a618m
Author: wenwen <wenwen-git@xxx.de>
Date:	Wed Nov 23 21:25:54 2016 +0100

    another test

```

my `.git/config` is not changed by `git-subrepo clean --ALL`, therefore, if I fetch changes, I will still get changes from "subrepotest", which is changed in "subrepotest" repo directly, not in my regular repo "testproject".

Then I manually removed the subrepo entry of my "testproject" `.git/config`. Then I can not fetch new changes from "subrepotest" repo, but the old history of "subrepotest", which is not pulled from my regular repo"testproject", stayes in my `gitk` of "testproject".

**Comment by grimmySwe** [comment]
@callalilychen Sorry for the late answer.

`git subrepo clean` will remove all internal branches and remotes used for subrepo. It will though not clear out the commit information that is stored in the main repo when you perform subrepo operations.

Please let me know if this was not the answer you was looking for!
